tests/php/test-amp-helper-functions.php

```diff
@@ -147,6 +147,7 @@ public function test_amp_bootstrap_plugin_amp_disabled() {
 	public function test_amp_init_migration() {
 		global $wp_actions;
 		remove_all_actions( 'init' );
+		remove_all_actions( 'admin_init' );
```

Comment:
In the [implementation](https://github.com/ampproject/amp-wp/pull/6042/files#diff-8fb0a549117c9fc2cf5d5945aaecaf5746ce3805facbf8a9d66dfa32bd3950f4R228-R229), the `init` action has been replaced by `admin_init`. Should the `remove_all_actions( 'init' );` be removed from here as well?

Reply:
I removed it at first but then I got a fatal error:

```
PHP Fatal error:  Cannot redeclare gutenberg_render_block_core_archives() (previously declared in /app/public/content/plugins/gutenberg/build/block-library/blocks/archives.php:17) in /app/public/content/plugins/gutenberg/build/block-library/blocks/archives.php on line 17
```

The reason is Gutenberg should be using `require_once` instead of `require`. I've opened a PR to fix that: https://github.com/WordPress/gutenberg/pull/31148

Reply:
However, it's also needed because of Gutenberg doing this:

```php
add_action( 'init', 'gutenberg_register_legacy_social_link_blocks' );
```

That causes a `_doing_it_wrong()` for `WP_Block_Type_Registry::register()` since it is attempting to re-register the block. So otherwise we'd need to remove those actions specifically, or remove the hooks.

Reply:
Sounds good, thanks!